pkg/function/interface.go

```diff
@@ -50,6 +51,12 @@ type MapHandler interface {
 	HandleDo(ctx context.Context, keys []string, datum Datum) Messages
 }
 
+// MapStreamHandler is the interface of map stream function implementation.
+type MapStreamHandler interface {
+	// HandleDo is the function to process each coming message.
+	HandleDo(ctx context.Context, keys []string, datum Datum) <-chan Messages
```

Comment:
Similarly, define the interface function as:

```golang
HandleDo(ctx context.Context, keys []string, datum Datum, ch <-chan Messages) 
```

Anyone who creates the channel is supposed to close the channel.

Reply:
Also, it should be `ch <-chan Message`, not `Messages`.

Reply:
Makes sense, with current logic inside `HandleDo` it has to use another go routine to pass the value to the channel, which is not very idiomatic.